docs/adr/adr_database-history.md

```diff
@@ -0,0 +1,150 @@
+
+## Decision Drivers
+
+The technological solution for the problem requires
+
+1. Simplicity: Our application is fairly small. We don't have capacities to neither dive into complex frameworks nor to maintain them nor to onboard new developers for them. The format of the recorded data should be clear such that querying and analysing data is straightforward.
+1. Extensibility: it should not be tricky if different kinds of data need to be recorded, or if existing data schemas change
+1. Robustness: we have to make sure that the data recorded represents the actual data at that point in time when it's read later
+1. Security: every change to the DB should be recorded. Recorded data should not be modifiable
+1. Disk space: is cheap and only of minor concern here
+1. Integration in both dropapp and boxtribute 2.0
+
+## Considered Options
+
+There are five options with an impact on different levels of the software.
+
+### Audit table
+
+This is a record of changes, tracking change type, change time, change description, name of the table that the change was made in, ID of the changed record, and possibly a structured payload. The current `history` table implements this concept.
+A variation is to have another table to record details for each change.
+
+*Advantages*
+
+- changes of data are recorded
+- simple and extensible
+- can be combined with database triggers to record any change on DB level
+
+*Drawbacks*
+
+- at a later point in time, the historical state of the data cannot be clearly derived: one can see that a field of a row has changed but other fields might have changed as well
+- dealing with data type conversion when storing/reading change payload ("old"/"new" value)
+
+### History tables (or: shadow tables)
+
+For every relevant table create a history table with identical schema and an additional `effective` date column. On every change in a tracked table, a copy of the updated row is inserted as new record into the history table.
+
+*Advantages*
+
+- change date and user are recorded
+- simple, especially querying (select by highest value of `effective` date column)
+- separation of data and history
+- can be combined with database triggers to record any change on DB level (no change in application code necessary then)
+- history tables should be insert-only; they cannot be modified then
+
+*Drawbacks*
+
+- extensibility costs: every change in the original table requires an update in the corresponding history table
+- the state of foreign key fields (i.e. tables that a history table depends on) has to be recorded as well to still be correct later
+
+### Insert-only tables
+
+Whenever a row would change, it is marked as inactive and copied into a new, active row.
```

Comment:
I guess we would also need to add an `effective` column in the related tables, right?

Reply:
yep